### ISSUE TITLE
Is there compatibility with .NET Standard | Class Library (.NET Standard) projects?

### ISSUE THREAD

**Issue by blmiles** [mono/SkiaSharp#1349]
Hello,

I have a Xamarin.Forms project that had the now deprecated "Shared" project in the solution. PCL projects are no longer supported either.

I've had to swap the "Shared" project over to a _.NET Standard | Class Library (.NET Standard) project_ and add all Nuget packages. This targets the .Net Core framework v2.1

One of my Forms uses the SKCanvasView where my users do some basic 2d drawing and I load a background image. This WAS all working with the "shared" project in place.

SkiaSharp plugin (_version 1.60.3 all the way to 2.80.0 preview 14_), installs, no issue, as does SkiaSharp.Views.Forms.

However, **SkiaSharp.Views** (all versions) cites the following issue and my drawing Form is no longer functional:
_Package 'SkiaSharp.Views 2.80.0-preview.14' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project target framework '.NETStandard,Version=v2.1'. This package may not be fully compatible with your project._

There are no build errors and the app runs but the SKCanvasView no longer displays or is not functional at all.

As Microsoft have now forced developers to use  the .Net Standard project as the "shared" library for Xamarin.Forms, one would think that the SkiaSharp components would follow suit and be compatible with said MS standard, especially if the primary SkiaSharp library and the SkiaSharp.Views.Forms are not citing any compatibility issues.

Can some please advise on how to resolve this issue?

Thank you

**Comment by mattleibow** [comment]
Make sure to install SkiaSharp.Views.Forms in all the projects, not SkiaSharp.Views. SVF brings the cross-platform views, the SV is just for native apps.

**Comment by blmiles** [comment]
Thanks Matt,
FYI, I find that version 1.60.3 works in all projects but anything above that throws a build error in iOS projects.
I tried all right to latest preview.14
When I get a chance I will update libs and send build errors I'm seeing. Right now just need it to work.

**Comment by mattleibow** [comment]
OK, cool. When you update, post the errors here and I'll have a look.

**Comment by mattleibow** [comment]
Did you manage to get things to work?

**Comment by mattleibow** [comment]
Closing this because SkiaSharp is compatible with .NET Standard 1.3+.

**Comment by blmiles** [comment]
Sorry, wasn't paying attention... Yes, working but only up to 1.60.3

Might have a chance soon to update incrementally and try again.

**Comment by blmiles** [comment]
Hello,

An update after updating SkiaSharp from 1.60.3 to 2.80.2
Solution is Xamarin.Forms (shared) and an iOS project and Android project. Android works just fine.

Also posted in this thread: https://github.com/mono/SkiaSharp/issues/855

Here's the error in iOS build:
Severity	Code	Description	Project	File	Line	Suppression State
Error		Error while loading assemblies: /Users/[user]/Library/Caches/Xamarin/mtbs/builds/[mycompany].iOS/2b8bb42efaa82e056d0923743e8a2829/bin/iPhoneSimulator/Debug/libSkiaSharp.dll.	[mycompany].iOS		


2>  /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/bin/mtouch @/Users/[user]/Library/Caches/Xamarin/mtbs/builds/[mycompany].iOS/2b8bb42efaa82e056d0923743e8a2829/obj/iPhoneSimulator/Debug/response-file.rsp "--gcc_flags=-Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker /Users/[user]/Library/Caches/Xamarin/mtbs/builds/[mycompany].iOS/2b8bb42efaa82e056d0923743e8a2829/obj/iPhoneSimulator/Debug/Entitlements.xcent" 
2>C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(885,3): error MT0009: Error while loading assemblies: /Users/[user]/Library/Caches/Xamarin/mtbs/builds/[mycompany].iOS/2b8bb42efaa82e056d0923743e8a2829/bin/iPhoneSimulator/Debug/libSkiaSharp.dll.

I have tried removing this from the iOS .proj file: `<PlatformTarget>x86</PlatformTarget>`

And I've tried adding this to the .proj file: `ExcludeAssets="runtimes"` in the skiaSharp reference(s).

If I regress to 1.60.3 it works just fine.

Any other potential fixes?

Thanks

**Comment by mattleibow** [comment]
@blmiles could you create a new issue so we can track it better?
Also, can you delete the ` /Users/[user]/Library/Caches/Xamarin/mtbs` folder and see if that fixes it? There might be some residue left behind. Or try a rebuild as well.

But, this should work, so mostly a rebuild should work.

**Comment by mattleibow** [comment]
Wait a sec. You must _not_ install SkiaSharp.Views into a .NET Standard project. That is platform specific.

In that new issue, could you submit a detail build log or a binlog? Check out this blog for steps: https://www.meziantou.net/exploring-the-msbuild-log-using-the-binary-log-viewer.htm

**Comment by blmiles** [comment]
Hi Matt,

Yes, I have not since you told me that before, but while trying to fix this, I saw a dependency to it in the Nuget package manager... so tried anyway. Un-installed everything again, cleaned, reinstalled ShiaSharp and ShiaSharp.Views.Forms ONLY. V 1.60.3 as that's the latest I can install, no later.

And as for that path/folder, it is on the mac and I cannot seem to get to it, at least cannot get to /Caches within /library
Was going to try do exactly what you suggest. Any ideas on that? 

I will look at submitting the binlog. Is there any way to do that privately and not expose some things here?
Will also create a new issue. It is similar to this as I've posted abaove https://github.com/mono/SkiaSharp/issues/855

Thanks
